File: 1. 단계별로_풀어보기/10~19/13. 정렬/10. 10814.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const N = +inputArr[0]
const infoArr = inputArr.slice(1).map((inputStr, idx) => {
  const [ageStr, nameStr] = inputStr.split(' ')

  return [+ageStr, nameStr, idx]
})

// 정렬
infoArr.sort((a, b) => {
  if (a[0] < b[0]) return -1
  else if (a[0] === b[0]) return a[2] - b[2]
  else return 1
})

// 출력
infoArr.forEach(info => {
  process.stdout.write(`${info[0]} ${info[1]}\n`)
})

File: 1. 단계별로_풀어보기/10~19/13. 정렬/10 10814.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())

infoArr = list([int(a), b, idx] for idx in range(N) for a, b in [input().strip().split()])

infoArr.sort(key=lambda x: (x[0], x[2]))

for info in infoArr:
  output(f"{info[0]} {info[1]}\n")